src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts

```diff
@@ -204,8 +226,15 @@ export class HorizontalPodAutoscaler extends KubeObject<
 export class HorizontalPodAutoscalerApi extends KubeApi<HorizontalPodAutoscaler> {
   constructor(deps: KubeApiDependencies, opts?: DerivedKubeApiOptions) {
     super(deps, {
-      objectConstructor: HorizontalPodAutoscaler,
       ...opts ?? {},
+      objectConstructor: HorizontalPodAutoscaler,
+      checkPreferredVersion: true,
+      // Kubernetes < 1.26
+      fallbackApiBases: [
+        "/apis/autoscaling/v2beta2/horizontalpodautoscalers",
+        "/apis/autoscaling/v2beta1/horizontalpodautoscalers",
+        "/apis/autoscaling/v1/horizontalpodautoscalers",
+      ],
```

Comment:
I don't think these are necessary since the group hasn't changed.

We should be careful though since if a `v3` comes out it might change the representation. Though we can add that safety in a future PR.